### DISCUSSION TITLE
Support Python custom repo dependencies

### DISCUSSION THREAD

**Discussion by CynanX** [renovatebot/renovate#26573] [Suggest an Idea]
### Tell us more.

If you have a requirements file as below, the custom repo. dependency is not checked by renovate:

```
boto3==1.34.15
jsonschema==4.20.0
sample-project @ git+https://gitlab.com/location/sample.git#1.1.0
```

The above will pull the tagged 1.1.0 from the repo. mentioned.

See here for some write up of the formats supported https://pip.pypa.io/en/stable/reference/requirement-specifiers/#requirement-specifiers

**Comment by rarkins** [comment]
The docs gives these examples:

```
pip @ https://github.com/pypa/pip/archive/22.0.2.zip
requests [security] @ https://github.com/psf/requests/archive/refs/heads/main.zip ; python_version >= "3.11"
```

But both of those are _files_ which are fetchable using a normal GET. The example by @CynanX is a git reference, which is not a simple GET. Does pip determine this based on the git+https prefix, i.e. `https://gitlab.com/location/sample.git#1.1.0` wouldn't work?

**Reply by CynanX** [reply]
Some schemes pip recognises can be seen on line 63 here: https://github.com/pypa/pip/blob/main/src/pip/_internal/vcs/git.py

**Reply by rarkins** [reply]
Converted to issue: https://github.com/renovatebot/renovate/issues/26612
